crates/starknet/src/blocks/mod.rs

```diff
@@ -54,6 +58,29 @@ impl StarknetBlocks {
         self.num_to_block.insert(block_number, block);
         self.last_block_hash = Some(hash);
     }
+
+    pub fn connect_state_diff_to_block(
+        &mut self,
+        block_number: BlockNumber,
+        state_diff: StateDiff,
+    ) {
+        self.num_to_state_diff.insert(block_number, state_diff);
+    }
+
+    pub fn get_by_block_id(&self, block_id: BlockId) -> Option<&StarknetBlock> {
+        match block_id {
+            BlockId::Hash(hash) => self.get_by_hash(Felt::from(hash)),
+            BlockId::Number(block_number) => self.num_to_block.get(&BlockNumber(block_number)),
+            BlockId::Tag(BlockTag::Latest) => {
+                if let Some(hash) = self.last_block_hash {
+                    self.get_by_hash(hash)
+                } else {
+                    None
+                }
+            }
+            BlockId::Tag(BlockTag::Pending) => None,
```

Comment:
[82e7e1f](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/103/commits/82e7e1fb275c8a00709296cb60874f4e488ae5cb)

Reply:
Better use `assert_eq!` instead of `assert!`

Reply:
this will lead to cascading changes, to use assert_eq! the arguments should implement Eq and Debug traits. StarknetBlock doesn't implement Debug, to implement it all of the members of StarknetBlock should implement Debug and Eq. Transactions doesn't implement Debug, so every variant of Transaction should be changed to implement Debug